### PR TITLE
Update README to reflect known issues

### PR DESCRIPTION
**Pull request by victorhazbun**
I was motivated to create this pull-request since I was having issues while trying to implement the library into my codebase. See the Github issue here: https://github.com/NoRedInk/elm-decode-pipeline/issues/37

### CODE REVIEW

**Comment by michaelglass** [comment]
Thank you for your PR!

I'm not sure this is the right place for this change. 

The error in question looks like this:

```
-- NAMING ERROR -------------------- ./path/to/your/File.elm

This usage of variable `required` is ambiguous.

67|                         |> required "hullabaloo" string
                               ^^^^^^^^
Maybe you want one of the following?

    Json.Decode.Pipeline.required
    Html.Attributes.required
```

There are lots of namespace collisions across packages in Elm. Take, for instance, `map` which is I think in every library there is. (Maybe.map, List.map, Dict.map, Set.map, etc, etc, etc).

This type of warning feels like it's a language / error message comprehension problem. Do you have any suggestions for how that error message could have been more clear?

File: README.md [hunk 1]
```diff
@@ -103,5 +103,13 @@ In this case, the result would be:
 }
 ```
 
+## Known Issues
```
Comment (michaelglass) [inline README.md:106]: a known issue implies that there is a understood bug. I don't think this is a bug

Comment (victorhazbun) [reply README.md:106]: You totally right.

**Comment by victorhazbun** [comment]
The error reported by Elm is a little confusing for Elm beginners, that's all. It will be nice if Elm can give you tips about how to resolve this kind of issues. And yes, this is not a bug on the elm-decode-pipeline lib, but again it will be nice to have documented this kind of scenarios somewhere in the library, maybe the README.md file is not the correct place but maybe in the Wiki page will be a good place. No?

File: README.md [hunk 2]
```diff
@@ -103,5 +103,13 @@ In this case, the result would be:
...
+
+- (Version 3.0.0) The `Json.Decode.Pipeline` lib exposes a function named `required` which is also exposed by the Elm `Html` library. To avoid compilation errors, consider aliasing the library import as follows:
+
+```elm
+import Json.Decode.Pipeline as JsonDecodePipeline exposing (decode, required, optional, hardcoded)
```
Comment (michaelglass) [inline README.md:111]: this alternative aliases the library but not `required`. e.g.

```elm
import Json.Decode.Pipeline as VeryNamespacedPipeline exposing (required)
import HtmlAttributes as EvenMoreNamespacedAttributes exposing (required)
```
I would still have a naming collision.

Some workarounds are
- not expose multiple functions sharing the same name
  - avoiding `exposing (..)` helps with that
- calling the function _including_ the namespace, e.g. `Html.Attributes.required` and `Json.Decode.Pipeline.required` will never collide

these are Elm language concerns, which is, IMO, outside of the scope of this library's readme.

**Comment by michaelglass** [comment]
I just think "less is more" in this regard. I appreciate the feedback and we'll keep it under consideration. If you have ideas around more helpful error messaging, the source is here: https://github.com/elm-lang/elm-compiler/blob/8a86d09f91c0b50141311309f72488074afed87a/compiler/src/Reporting/Error/Canonicalize.hs#L976

**Comment by michaelglass** [comment]
another place to report error messages is https://github.com/elm-lang/error-message-catalog/issues

**Pull request closed without merging**